Sources/DolbyIORTSCore/StreamCoordinator.swift

```diff
@@ -192,105 +228,106 @@ open class StreamCoordinator {
+    func startSubscribe() async -> Bool {
+        async let startSubscribeStateUpdate: Void = stateMachine.startSubscribe()
+        async let startSubscribe = subscriptionManager.startSubscribe()
+        let (_, success) = await (startSubscribeStateUpdate, startSubscribe)
+        return success
+    }
+}
+
 // MARK: SubscriptionManagerDelegate implementation
 
 extension StreamCoordinator: SubscriptionManagerDelegate {
     public func onSubscribedError(_ reason: String) {
-        Task {
+        let task = Task {
             await stateMachine.onSubscribedError(reason)
         }
+        coordinatorTasksContinuation?.yield(task)
     }
 
     public func onSignalingError(_ message: String) {
-        Task {
+        let task = Task {
             await stateMachine.onSignalingError(message)
         }
+        coordinatorTasksContinuation?.yield(task)
     }
 
     public func onConnectionError(_ status: Int32, withReason reason: String) {
-        Task {
+        let task = Task {
             await stateMachine.onConnectionError(status, withReason: reason)
-            if Self.configuration.retryOnConnectionError {
-                taskScheduler.scheduleTask(timeInterval: Self.configuration.retryTimeInterval) { [weak self] in
-                    guard let self = self else { return }
-                    Task {
-                        switch await self.stateMachine.currentState {
-                        case let .error(state):
-                            switch state.error {
-                            case .connectFailed:
-                                self.taskScheduler.invalidate()
-                                _ = await self.connect(streamName: state.streamDetail.streamName, accountID: state.streamDetail.accountID)
-                            default:
-                                break
-                            }
-                        default:
-                            break
-                        }
-                    }
-                }
-            }
         }
+        coordinatorTasksContinuation?.yield(task)
     }
 
     public func onStopped() {
-        Task {
+        let task = Task {
             await stateMachine.onStopped()
         }
+        coordinatorTasksContinuation?.yield(task)
     }
 
     public func onConnected() {
-        Task {
+        let task = Task {
             await stateMachine.onConnected()
-            if Self.configuration.subscribeOnSuccessfulConnection {
-                _ = await startSubscribe()
-            }
+            _ = await startSubscribe()
         }
+        coordinatorTasksContinuation?.yield(task)
     }
 
     public func onSubscribed() {
-        Task {
+        let task = Task {
             await stateMachine.onSubscribed()
         }
+        coordinatorTasksContinuation?.yield(task)
     }
 
     public func onVideoTrack(_ track: MCVideoTrack, withMid mid: String) {
-        Task { [weak self] in
+        let task = Task { [weak self] in
```

Comment:
Why does this one need `[weak self]` but the others?

Reply:
While its good to capture [weak self] in closures, our StreamCoordinator is a singleton and [weak self] is not required here.

Reply:
Shall this one be removed?

Reply:
I suggest keep it ? Mainly because its a clean coding practice to capture weak self in escaping closures

Reply:
Why does this escaping closure is different from others? Such as lines 299 to 301

Reply:
No its not different from others, I have added [weak self] on all Task closures now